src/stcal/tweakreg/tweakreg.py

```diff
@@ -253,14 +254,43 @@ def _parse_refcat(abs_refcat: str | Path,
         )
 
     if Path(abs_refcat).is_file():
-        return Table.read(abs_refcat)
+        return _parse_sky_centroid(Table.read(abs_refcat))
 
     msg = (f"Invalid 'abs_refcat' value: {abs_refcat}. 'abs_refcat' must be "
            "a path to an existing file name or one of the supported "
            f"reference catalogs: {_SINGLE_GROUP_REFCAT_STR}.")
     raise ValueError(msg)
 
 
+def _parse_sky_centroid(catalog: Table) -> Table:
+    """Turn SkyCoord object into simple RA/DEC columns.
+
+    The inclusion of SkyCoord objects via sky_centroid.ra and sky_centroid.dec
+    permits the use of catalogs directly from the jwst source_catalog step.
+    No action is taken if the catalog already contains RA and DEC columns.
+    """
+    cols = [name.lower() for name in catalog.colnames]
+    if ("ra" in cols) and ("dec" in cols):
+        if "sky_centroid" in cols:
+            msg = ("Catalog contains both (RA, DEC) and sky_centroid. "
```

Comment:
is it `contains both (RA, DEC)` or maybe `contains both (ra, dEc)`?

Reply:
I updated the error message a bit, let me know if you are ok with the change